[oneplus] Add OnePlus devices

paulverbeke opened the pull request:
Since the only "official" updates policy is a community post here https://community.oneplus.com/thread/1462181 (and here for the old one https://community.oneplus.com/thread/862347) should this be considered enough to start a PR for OnePlus devices ?
And since these are only promises should I display the exact calculated date or just yes ?
For instance for the active support they don't say X years of support but X major Android updates ? Should I translate these 4 major Android updates to 4 years ?

paulverbeke committed:
[new product] Add OnePlus devices

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+---
+title: OnePlus
+category: device
+tags: oneplus 1plus mobile-phone
```
```suggestion
tags: mobile-phone oneplus
```

Remove `1plus` as the brand name is OnePlus (https://en.wikipedia.org/wiki/OnePlus). Also sort alphabetically.

paulverbeke replied:
no worries, it's done !

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+
+releases:
+-   releaseCycle: "12"
+    releaseLabel: "OnePlus 12"
```
```suggestion
    releaseLabel: "OnePlus 12"
```

Setting `releaseLabel: "OnePlus __RELEASE_CYCLE__"` could be used at the upper level (just after the `releasePolicyLink`) to avoid having to set it for all release cycle.

paulverbeke replied:
yes good idea. done !

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+releases:
+-   releaseCycle: "12"
+    releaseLabel: "OnePlus 12"
+    releaseDate: 2023-12-05
```
```suggestion
    releaseDate: 2023-12-11 # https://www.gsmarena.com/oneplus_12-12725.php
```

According to https://www.gsmarena.com/oneplus_12-12725.php (2023-12-05 is the announcement, not the availability date).

Also adding the link to the source of each date would be helpful for future maintainers IMO.

paulverbeke replied:
that depends what we want to use. If we look in the blog https://oxygenupdater.com/article/396/ that is much more thorough (even with linked community posts, announcements and social networks) we see that announced date and released date are the same at OnePlus.
Moreover GSMarena is not even providing their source for the dates ! Here the 12 and 12R are announced together (https://community.oneplus.com/thread/1514801169317232648) but GSMarena have 2 months gap.

So should we use "announced and released in china" (2023-12-05) or "announced and released globally" (early 2024, apparently 2024-01-23 https://community.oneplus.com/thread/1514801169317232648) ?
For now I choose global dates, found on oneplus community post.

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+    support: 2027-12-05 # approximation "4 major Android updates"
+    eol: 2028-12-05
+    discontinued: false
+    link: https://www.gsmarena.com/oneplus_12-12725.php
```
```suggestion
    link: https://www.oneplus.com/launch/12
```

We avoid using direct gsmarena links on endoflife.date for now, as we try to not advertise sites and usually prefer "official" links (such as https://www.oneplus.com/launch/12). If you don't agree it is debatable, but I would prefer a discussion on https://github.com/endoflife-date/endoflife.date/discussions rather than here.

paulverbeke replied:
As you prefer, but earlier you said

> And for the link attribute prefer using the support site (such as [service.oneplus.com/us/search/search-detail?id=2123192&articleIndex=2](https://service.oneplus.com/us/search/search-detail?id=2123192&articleIndex=2)) than the "marketing" site (such as [oneplus.in/11r](https://www.oneplus.in/11r)). Link to support sites have less chances to change over time.

And you were right: OnePlus is releasing a lot of phones and isn't keeping the marketing pages alive long.
But at the same time I can't seem to find support pages like the one you found, so I went with the next best thing: good static reference pages.

captn3m0 replied:
There's also a oneplus.com/global/support pages, which seem reasonable and not-dead. For eg https://www.oneplus.com/global/support/spec/oneplus-2

The list at https://www.oneplus.com/global/support/spec seems comprehensive enough, at least for the phones.

paulverbeke replied:
great find thanks that looks good to me. Will use this

paulverbeke replied:
except there's only old models OnePlus One through OnePlus 9. Missing all new ones. Anyway it's still a good reference when I'll add old models. But for now I'll stay on gsmarena for newer ones (and maybe marketing pages for the very last ones that are still currently sold, if you really want that ?)

captn3m0 replied:
Current links are good, and we can use the support/spec links for older devices. resolving this.

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+    position: after-release-column
+    label: Supported Android
+    description: Supported Android versions range
+    link: https://endoflife.date/android
```
```suggestion
customColumns:
-   property: supportedOxygenOSVersions
    position: after-release-column
    label: Supported OxygenOS
    description: Supported OxygenOS versions range
    link: https://wikipedia.org/wiki/OxygenOS
```

I think it would makes more sense to refer to OxygenOS rather than Android, as the support for OxygenOS may be different than Android.

paulverbeke replied:
I get it, but it would be basically the same problem with pixel and fairphone: in the custom column we are talking about the support of their own OSes. Fairphone is running Fairphone OS and Pixel is running their own flavor of Android (that doesn't have a name AFAIK ?). Fairphone OS is not a big derivative from AOSP but Pixel OS definitely adds a lot over Android.
Moreover OxygenOS version wasn't always corresponding with Android so could be source of confusion here. 
![image](https://github.com/endoflife-date/endoflife.date/assets/6412153/eb1b4c6d-fb2a-4d9e-98fe-5a5ba53720a9)

IMHO it's either we change the column on all 3 pages or none of them.

captn3m0 replied:
My preference would be to stick to OxygenOS. We're trying to [Be Helpful](https://github.com/endoflife-date/endoflife.date/wiki/Guiding-Principles#be-helpful) to the end-user. From their perspective, the page should answer "What is the highest version of my OS I can upgrade to". Using OxgenOS numbering clarifies the same, since that is what OnePlus uses in their settings screen, and upgrade popups, and documentation?

Google still calls Pixel updates as Android updates, that seems okay.

>  Moreover OxygenOS version wasn't always corresponding with Android so could be source of confusion here.

This is more the reason to document it as OxygenOS version numbers - users reading an Android version number might be confused as well. We can perhaps add an equivalence table/link.

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+# Because OnePlus is using community forum posts as official releases posts it is hard to parse the last Android/OxygenOS releases 
+# for all devices so I'm using the unofficial https://oxygenupdater.com/news/all/ that have been collecting all OnePlus OxygenOS announcements for several years and is an established and trusted way to get OP updates.
+# still not great but easier to search and parse than OnePlus Community and more up to date than GsmArena
+# There is also https://www.androidupdatetracker.com/policy/oneplus that is easier to parse and up to date but AFAIK is less established
```
Didn't know those websites, thanks for the explanation.

What do you mean by saying https://www.androidupdatetracker.com/ is "less established" ? Is it incomplete / less up-to-date than oxygenupdater / sometime inaccurate ? Or just less popular ?

I ask because it seems it would be simpler for maintainers to refer to https://www.androidupdatetracker.com/ rather than https://oxygenupdater.com/, so if possible I would prefer if it was used as the reference.

paulverbeke replied:
What I meant is that I can't attest to the seriousness of https://www.androidupdatetracker.com/ because I saw at least one mistake in their oneplus policy here https://www.androidupdatetracker.com/policy/oneplus regarding the Nord N series. I have no info into their research and the site is mentioned nowhere. But they have good links to release announcements of major versions. https://oxygenupdater.com/ have correct data (so far) even helping me see mistakes GSMarena had done in release dates. I saw them mentioned several times in blogs, and they have an open source app on github that shows that they've been present for 4-5 years.

captn3m0 replied:
I've moved these links to the description, because they might be helpful to end users as well. Didn't feel the nuance about Nord N was worth adding for now (Nord N isn't being tracked for now as well).

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+OnePlus phones receives updates every two months.
+
+The company is well known for its flagships phones which cover the global market and are the very top end of the brand in price and performance (ex: 10T, 10 Pro, 12, etc.). Then you have the India-focused R-series (one exception to this rule is the OnePlus 12R, in December 2023 [they announced](https://oxygenupdater.com/article/436/) that the 12R will not just launch in India, but North America a ...
+Finally there is the mid-range Nord/Nord CE series (ex: Nord 3, Nord CE 3, etc.), and the entry-level Nord N series (Nord N300, Nord N30, etc.).
```
This first paragraph feels a bit too much commercial / descriptive to me and don't give much information about the support / EOL policy. Do you think you could drop it ?

paulverbeke replied:
yes completely you are right ! done !

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+
+In June 2018, the first update policy [announcement](https://community.oneplus.com/thread/862347) was made offering two major Android updates and three years of security updates for the OnePlus 3/3T, 5/5T, 6 and onwards.
+
+Then in July 2021, OnePlus [announced](https://community.oneplus.com/thread/1462181) that following the merging of OPPO developpers with theirs and ColorOS with OxygenOS, they could now offer more years of software updates to their existing lineups of phones. They now offer three major Android updates and four years of security updates for the flagship line (including T/R) starting from OnePlus 8 ...
```
Is it Android updates or OxygenOS updates ?

paulverbeke replied:
OnePlus is talking everywhere specifically about "major Android updates"

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+
+Then in July 2021, OnePlus [announced](https://community.oneplus.com/thread/1462181) that following the merging of OPPO developpers with theirs and ColorOS with OxygenOS, they could now offer more years of software updates to their existing lineups of phones. They now offer three major Android updates and four years of security updates for the flagship line (including T/R) starting from OnePlus 8 ...
+
+In November 2022, it was [announced](https://community.oneplus.com/thread/1211291251581124608) that starting from 2023 OnePlus will offer four generations of OxygenOS (Android versions) and five years of security updates on selected devices. In practice this policy will apply to their flagship lineup starting from OnePlus 11 onwards.
```
Is there OxygenOS versions not based on Android ?

paulverbeke replied:
They are. Do you think my sentence is confusing ?

captn3m0 replied:
I think the new description resolves this.

marcwrobel commented on products/oneplus.md:
```diff
@@ -0,0 +1,103 @@
+
+> OnePlus is a Chinese manufacturer of Android phones and other consumer electronics.
+
+OnePlus phones receives updates every two months.
```
Are all OnePlus phones running OxygenOS ? If so I think it would be useful to add a paragraph at the start of the description about the relation between OxygenOS and Android, and then never mention Android again.

paulverbeke replied:
I've added a small sentence to explain. Regarding mentioning Android or OxygenOS throughout the page, see my answer on the custom column conversation

captn3m0 commented on products/oneplus.md:
```diff
@@ -0,0 +1,119 @@
+
+Then in July 2021, OnePlus [announced](https://community.oneplus.com/thread/1462181) that they would now offer three major Android updates and four years of security updates for the flagship line (including T & R series) starting from OnePlus 8 onwards, two major Android updates and three years of security updates for the Nord/Nord CE series, and finally one major Android update and three years o ...
+
+In November 2022, OnePlus [announced](https://community.oneplus.com/thread/1211291251581124608) that they will offer "four generations of OxygenOS" (Android versions) and five years of security updates on "selected devices". In practice this policy will apply only to their flagship lineup (including T & R series) starting from OnePlus 11 onwards.
```
+1 to what @marcwrobel suggested:

> Overall that's a good start but I think there is too much information in the description.

Suggestion to implement that:

```suggestion
OnePlus phones run OxygenOS, which is based on Android. It receives updates every two months.

OnePlus supports [four generations of OxygenOS](https://community.oneplus.com/thread/1211291251581124608) (Android versions) and five years of security updates on "selected devices". In practice this policy applies only to their flagship lineup (including T & R series) starting from OnePlus 11 onwards.
```

> this policy applies only to their flagship lineup

It is confusing whether "this policy" means here, in context of Non-Flagship devices:

1. Four generations of OxygenOS are supported but 5 years of security updates are not guaranteed.
2. No support guarantees are made.

Minor note on tone: I changed the continuous tense to present. https://github.com/endoflife-date/endoflife.date/wiki/Style-Guide#tone-and-text has more examples.

paulverbeke replied:
Yes, their update policies are changed and announced all over the place, that's why there's several links. It's a mess. 
In my last force push I used your suggestion (i.e. only mention current update policy and disregard old ones) to cut down on text length, but still mentioned the policies of the other 3 range products (Nord, Nord CE, Nord N).
I can cut even more information but that'll mean completely removing a policy

captn3m0 replied:
Updated description looks good 👍🏽

captn3m0 committed:
[oneplus] Minor changes

- removed comments, we can track TODO elsewhere.
- Dropped "chinese" mention. We only use this in one other place (Motorola), will drop from there as well.

captn3m0 commented on products/oneplus.md:
```diff
@@ -0,0 +1,66 @@
+---
+title: OnePlus
```
Not very happy with the title, but I don't like "OnePlus Phone" either. Anyone has suggestions?

paulverbeke replied:
I already see "Fairphone", "Google Pixel", "Samsung Mobile" (🤮)...
So looks fine to me like this.
Or eventually "OnePlus phones" if you really want to change it

paulverbeke replied:
@captn3m0 do we leave OnePlus ? Or can we settle on something else ? so that we can merge this

captn3m0 replied:
Let's leave it at OnePlus for now.

paulverbeke committed:
Update products/oneplus.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

paulverbeke committed:
fix supportedOxygenOSVersions

marcwrobel reviewed (approved):
LGTM, thanks a lot for this contribution (and your patience !).
